3ds Max/Max2Babylon/Exporter/BabylonExporter.Mesh.cs

```diff
@@ -1342,27 +1346,41 @@ public void GenerateCoordinatesAnimations(IIGameNode meshNode, List<BabylonAnima
         /// <param name="tangent"></param>
         /// <param name="bitangent"></param>
         /// <returns>
-        /// -1 when the normal is not flipped
-        /// 1 when the normal is flipped
+        /// 1 when the bitangent is nearly 0,0,0 or is not flipped
+        /// -1 when the bitangent is flipped (oposite direction of the cross product of normal ^ tangent)
         /// </returns>
-        private int GetW(IPoint3 normal, IPoint3 tangent, IPoint3 bitangent)
+        private float GetW(IPoint3 normal, IPoint3 tangent, IPoint3 bitangent)
         {
-            //Cross product bitangent = w * normal ^ tangent
-            float x = normal.Y * tangent.Z - normal.Z * tangent.Y;
-            float y = normal.Z * tangent.X - normal.X * tangent.Z;
-            float z = normal.X * tangent.Y - normal.Y * tangent.X;
+            float btx = Math.Abs(bitangent.X) <= Tools.Epsilon ? 0 : bitangent.X;
+            float bty = Math.Abs(bitangent.Y) <= Tools.Epsilon ? 0 : bitangent.Y;
+            float btz = Math.Abs(bitangent.Z) <= Tools.Epsilon ? 0 : bitangent.Z;
 
-            int w = Math.Sign(bitangent.X * x);
-            if (w == 0)
-            {
-                w = Math.Sign(bitangent.Y * y);
-            }
-            if (w == 0)
+            if( btx == 0 && bty == 0 && btz == 0)
             {
-                w = Math.Sign(bitangent.Z * z);
+                return 1;
             }
 
-            return w;
+            float nx = Math.Abs(normal.X) <= Tools.Epsilon ? 0 : normal.X;
```

Comment:
Maybe add an override for `Tools.IsAlmostEqualTo`?

Reply:
Well technically this is
float nx = Tools.IsAlmostEqualTo(normal.X,Tools.Epsilon)? 0 : normal.X;
so i do not know how to name this function :-)

Reply:
Ok, no worries. I just like using helper functions for math, so they're reusable and documented via code.

A header like:
```
public static bool IsAlmostEqualTo(this float current, float other, float epsilon)
{
    return Math.Abs(current - other) < epsilon
}
```

Should let us use it like

```
float nx = current.isAlmostEqualTo(0, Tools.Epsilon);
```

Just to minimize the amount of wild math we're using